inference-engine/src/gna_plugin/gna_graph_compiler.cpp

```diff
@@ -1303,8 +1276,9 @@ void GNAGraphCompiler::AffinePrimitive(InferenceEngine::CNNLayerPtr layer, bool
     auto inputPrecision = quantized ? Precision(Precision::I16) : inputs->getPrecision();
 
     auto input_data = HasTo2DReshapeData(layer) ? Get2DReshapedData(inputs, 8) : inputs;
-    uint32_t num_rows_in = FROM_IR_DIM(input_data, 1);
-    uint32_t num_columns_in = FROM_IR_DIM(input_data, 2);
+    auto in_dims = input_data->getDims();
+    uint32_t num_rows_in = InferenceEngine::details::product(in_dims) / in_dims.front();
+    uint32_t num_columns_in = in_dims.front();
```

Comment:
FROM_IR_DIM(input_data, 1) and FROM_IR_DIM(input_data, 2) will get correct results for 2D inputs only. The change was done to support 4D inputs that are possible for example in ScaleShift with only second dimension not equal to 1. So the changed values will be the same as old ones only for 2D tensors. I'm not sure that this assert will be useful since FROM_IR_DIM macro gets access to the same input_data->getDims() data.